### DISCUSSION TITLE
Adding Tuya Local

### DISCUSSION THREAD

**Discussion by rhocking73** [make-all/tuya-local#2488] [Q&A]
I am trying to add my Holman WX1 irrigation to Home Assistant. I have added the Tuya-Local integration via HACS. I have then clicked to add the integration. I get the popup asking if I want to use the Smart Life cloud assisted device setup. I click this and the add device pop up comes up populated with the name of my Tuya Wifi Hub. As this is the hub I have left the hub question as none. I have then pressed next. The device IP screen comeup saying I may need to locate the IP address. The next pop up comes up requesting  the device ID, the IP address, the Local Key and the protocol version. I have gone to the TUYA developers platform, linked my Tuya Smart Ap. I have a list of the 4 devices in my account - The wifi hub, a wifi smart socket and the two wifi tap timers. I have copied the device ID of the wifi hub and then in API explorer I have queried the device id of the wifi hub and from this I have the Device ID, the Local Key and the UUID of the wifi hub. I have entered the device ID of the hub, the local key and I have found the IP address of the Tuya hub on my router. (This is a different IP address to the IP address listed in the API explorer entry for the Wifi Hub). It comes up with protocal 3.3, however I have also tried auto. I then click submit. This process returns an error "Unable to connect to your device with those details. It could be an intermittent issue, or they may be incorrect."

I have checked the logs in the Home assistant core and the error reads:

Logger: custom_components.tuya_local.device
Source: custom_components/tuya_local/device.py:464
integration: Tuya Local ([documentation](https://github.com/make-all/tuya-local), [issues](https://github.com/make-all/tuya-local/issues))
First occurred: xxxxxxx (28 occurrences)
Last logged: xxxxxx

Test protocol error 901: Network Error: Unable to Connect

I have tried using the IP address for the wifi hub I have taken from my router, and also the IP address as it is recorded in the Tuya API Explorer query. Both result in the same error.

Any clue what I am doing incorrectly to get this integration to work? I think I am following all the instructions. I have ensured that the Tuya Smart Ap is closed when I am setting this up as there was a suggestion that the devices only want to connect to one local control at a time.

The standard Tuya integration is able to find the four devices, however when I use this integration, it finds four devices and creates 1 entity. The smart wifi socket has a contol however the tap timers in the control section states "This device has no entities". Clearly the Tuya integration is not working - hence trying to get the Tuya Local integration to work for me. Any advice?
Thanks in advance.

**Comment by make-all** [comment]
If the hub has no features of its own, then it will not be able to connect as a device in Tuya Local.  You need to go straight to adding a subdevice, choosing your hub.

It is difficult to tell if you are using the cloud assisted flow, as you start off explaining it, then jump to entering details from API Explorer.  If you use the cloud assisted flow, those should get populated automatically.

**Reply by make-all** [reply]
There are however already existing threads about the WX1 not working for others, so it seems it is a difficult/impossible device to get working.

**Reply by rhocking73** [reply]
Thanks for your advice. I am using the cloud assisted flow and the details are self populated. I first came across a different version of the Tuya local integration that was not the one that you re-worked. In this version, you had to find the the details using the API explorer. Because I couldn't get it to work I was searching on Github for references that mentioned the Holman WX 1 rather than just generic Tuya IOT devices and I came across this version of Tuya Local- so I uninstalled the previous version of Tuya Local and then tried this version. Could previous versions of the programme be messing with the installation? I'm new to Home Assistant, so perhaps I haven't deleted the previous integration properly.

I have followed your suggestion and tried to ad one of the WX1s using them as a sub device of the Wifi Hub - all the entries were prepolulated for me, and I get the same error. 

Grateful for any other suggestions about what to do. May be a problem product and I may have to look at another alternative for irrigation or just accept that I can't get it onto Home Assistant.
